process_sample_data.py

```diff
@@ -20,9 +21,9 @@ class ProcessSampleData():
     """
 
     def __init__(self, path_to_data):
-        self._flow_data_file = open(path_to_data, "r")
-        self._parseData()
-        self._flow_data_file.close()
+        with open(path_to_data, "r") as self._flow_data_file:
+            self._parse_data()
+            self._remove_repeat_data()
```

Comment:
I gather that this is for testing, so efficiency isn't a huge concern, but you could save yourself some time if you dedup while you parse. You could reformulate the inners of `_parse_data` as a stream of tuples, keep track of previous and only append to the data lists if the new values are meaningfully different.